internal/controllers/cluster/cluster_controller_test.go

```diff
@@ -172,6 +172,50 @@ func TestClusterReconciler(t *testing.T) {
 		}, timeout).Should(BeTrue())
 	})
 
+	t.Run("Should successfully patch a cluster object if the spec diff is empty but the status conditions diff is not", func(t *testing.T) {
```

Comment:
Can you provide some context what use case this test is testing?

Reply:
This seems to be some other controller also patching InfrastructureReady to true in parallel to the cluster controller?

Reply:
I only added it for completeness, seems this really tests the patch helper?

Reply:
The same test, but for `cluster.Status.InfrastructureReady = true` is above this one

Reply:
Ah, I didn't realize this is ~ a copy of an existing test :D